faker/providers/__init__.py

```diff
@@ -173,20 +173,23 @@ def numerify(self, text='###'):
             text)
         return text
 
-    def lexify(self, text='????'):
+    def lexify(self, text='????', letters=None):
         """
         Replaces all question mark ('?') occurrences with a random letter.
 
         :param text: string to be parsed
+        :param letters: a set of letters to choose from.
         :returns: string with all letter placeholders filled in
         """
+        if letters:
+            return _re_qm.sub(lambda x: self.random_element(letters), text)
```

Comment:
rather than modifying `radom_element` and switch methods, I'd modify `random_letter` to accept a `letters` that defaults to `string.ascii_letters`